Dart/src/com/jetbrains/lang/dart/resolve/DartResolver.java

```diff
@@ -129,6 +123,16 @@ else if (leftReference != null) {
   }
 
   @Nullable
+  public PluginNavigationRegion findRegion(final PsiFile refPsiFile, final int refOffset, final int refLength) {
+    final VirtualFile refVirtualFile = refPsiFile.getVirtualFile();
```

Comment:
to get virtual file from PsiElement or PsiFile use DartResolveUtil.getRealVirtualFile(). I do not know exactly in which cases it works better than just PsiFile.getVirtualFile(), but for safety I prefer to use it.